examples/tide/Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM opensuse/leap:latest
+MAINTAINER wbrown@suse.de
+
+EXPOSE 8080
+
+RUN mkdir /src
+WORKDIR /src
+
+ADD ./ /src/
+
+RUN zypper in -y gcc libopenssl-devel openssl wget && \
+    wget https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init && \
+    chmod +x rustup-init && \
+    ./rustup-init -v -y && \
+    source /root/.profile && \
+    cargo build --example actix --release && \
+    zypper rm -y gcc libopenssl-devel && \
+    rm -r rustup-init /root/.rustup
+
```

Comment:
There is a way to build this in an intermediate container and then copy the binary over. Have a look at https://github.com/kanidm/kanidm/blob/master/kanidmd/Dockerfile#L34

Reply:
Hopefully I pattern matched this correctly—I'm not experienced with Docker.

Reply:
It's *prettttyyyyyyyyy* close to what it needs. At this point I'm happy for you to leave it as is, and I will fix it later :)

Reply:
I tried running it from the tide directory with `docker build .` and it seems to only include the example directory, rather than the whole repo (which is what the `cargo build` command seems to suppose). Because the actix dockerfile had an identical add command, I take it I'm building the container incorrectly?

Reply:
I think the file should look like this:

```
ARG BASE_IMAGE=opensuse/leap:latest
FROM ${BASE_IMAGE} AS builder
MAINTAINER wbrown@suse.de

RUN mkdir /src
WORKDIR /src

ADD ./ /src/

RUN zypper in -y gcc libopenssl-devel openssl wget && \
    wget https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init && \
    chmod +x rustup-init && \
    ./rustup-init -v -y && \
    source /root/.profile && \
    cargo build --example tide

FROM ${BASE_IMAGE}
LABEL mantainer william@blackhats.net.au

RUN zypper ref && \
    zypper install -y \
        timezone && \
    zypper clean -a

COPY --from=builder /src/target/debug/examples/tide /sbin/

EXPOSE 8080
CMD ['/sbin/tide']

```

Reply:
I tried something like this and it almost works—I also had to copy over the static directory though. Ultimately I didn't get anything on localhost:8000 when I tried to run the container but I'm trying to see if that can be fixed simply.

Reply:
I think maybe add

```
COPY --from=builder /src/static /static
WORKDIR /
```

That may fix it.